File: 프로그래머스/lv2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    answer = ""
    flag = 0        # 공백이 나오면 플래그 0
    
    for w in s:
        if flag:
            if w == " ":    # 단어 후 공백 등장: flag down
                flag = 0
                answer += w
            else:
                answer += w.lower()
            continue
            
        if flag == 0:
            if w == " ":
                answer += w
            else:
                flag = 1    # 공백 다음 첫 문자: 플래그 on, 대문자 처리
                answer += w.upper()
            continue
    
    return answer

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
class Solution {
    public String solution(String s) {
        String answer = "";
        StringBuilder sb = new StringBuilder();
        boolean flag = false;
        
        s = s.toLowerCase();
        
        for (char a : s.toCharArray()) {
            if (flag == false && a != ' ') {        // 첫 글자라면 대문자로 바꿔서 추가
                a = Character.toUpperCase(a);
                flag = true;
            } else {
                if (flag == true && a == ' ') {  // 어절 끝나면 flag 바꿈
                flag = false;
                }
            }
            sb.append(a);    
        }
        
        answer = sb.toString();
        
        return answer;
    }
}